source/DataWriter.cpp

```diff
@@ -56,21 +50,70 @@ DataWriter::~DataWriter()
 
 
 // Save the contents to a file.
-void DataWriter::SaveToPath(const std::string &filepath)
+void DataWriter::SaveToPath(const string &filepath)
 {
 	Files::Write(filepath, out.str());
 }
 
 
 
+// Save the contents to a stream.
+void DataWriter::SaveToStream(ostream &stream)
+{
+	stream << out.str();
+}
+
+
+
+// Pass the contents to a function.
+void DataWriter::SaveToFunction(const function<void(string)> func)
+{
+	func(out.str());
+}
+
+
+
+// Writes the contents of the string without any escaping, quoting or
+// any other kind of modification.
+DataWriter &DataWriter::WriteRaw(const string &c)
+{
+	out << c;
+	return *this;
+}
+
+
+
+// Raw write for numeric types.
+template <class C>
+DataWriter &DataWriter::WriteRaw(const C &c)
```

Comment:
This should be in the header with the other templates :)

Reply:
It is, but if I remove the template from here, I get a compiler error.

`‘C’ does not name a type`

Reply:
> It is

sry I don't get it. it's in the cpp file currently. And why remove the template?

Reply:
https://github.com/endless-sky/endless-sky/pull/8725/files#diff-99518e0fcab68a0387d2338e9bda83c81d6f5e691d182589a10a10276773fb43R51-R52

Reply:
https://github.com/endless-sky/endless-sky/pull/8725/files#diff-5f5b7984c40c9b8f96d4812489f060eb5a1a920bcb124d4385d23e4cb5390e65R52-R60 move this in the header